File: practice_problems/maxFrequencyInt.js

input = [-2, -2, -2, -2, 3, 5];

function maxFrequencyInt(input) {

    let integerFrequencies = {};

    mostOccuringInt = null;
    count = 0;

    for(let i = 0; i < input.length; i ++) {
        if (integerFrequencies[input[i]] !== undefined) {
            integerFrequencies[input[i]] += 1;

        } else {
            integerFrequencies[input[i]] = 1

        }

        if (integerFrequencies[input[i]] > count) {
            count = integerFrequencies[input[i]];
            mostOccuringInt = input[i];
        }
    }

    return mostOccuringInt

}

console.log(maxFrequencyInt(input));

File: practice_problems/max_frequency_int.py
"""
Maximum frequency integer
"""


def max_frequency_int(input):
    """
    >>> max_frequency_int([-2, -2, -2, -2, 4, 5])
    -2
    >>> max_frequency_int([])
    """

    if len(input) == 0:
        return

    int_frequencies = {}

    most_occurring_int = input[0]
    max_count = 0

    for integer in input:
        int_frequencies[integer] = int_frequencies.get(integer, 0) + 1
        if int_frequencies[integer] > max_count:
            max_count = int_frequencies[integer]
            most_occurring_int = integer

    return most_occurring_int





